modules/loader/src/main/java/com/opengamma/strata/loader/csv/RatesCalibrationCsvLoader.java

```diff
@@ -0,0 +1,368 @@
+      curveNodes.add(createCurveNode(typeStr, conventionStr, timeStr, label, quoteKey, spread));
+    }
+    return buildCurveDefinition(settingsMap, allNodes);
+  }
+
+  // build the curves
+  private static Map<CurveName, NodalCurveDefinition> buildCurveDefinition(
+      Map<CurveName, LoadedCurveSettings> settingsMap,
+      Map<CurveName, List<CurveNode>> allNodes) {
+
+    ImmutableMap.Builder<CurveName, NodalCurveDefinition> results = ImmutableMap.builder();
+    for (Map.Entry<CurveName, List<CurveNode>> entry : allNodes.entrySet()) {
+      CurveName name = entry.getKey();
+      LoadedCurveSettings settings = settingsMap.get(name);
+      if (settings == null) {
+        throw new IllegalArgumentException(Messages.format("Missing settings for curve: {}", name));
+      }
+      results.put(name, settings.createCurveDefinition(entry.getValue()));
+    }
+    return results.build();
+  }
+
+  // create the curve node
+  private static CurveNode createCurveNode(
+      String typeStr,
+      String conventionStr,
+      String timeStr,
+      String label,
+      QuoteKey quoteKey,
+      double spread) {
+
+    if ("FIX".equals(typeStr) || "IborFixingDeposit".equals(typeStr)) {
+      return curveIborFixingDepositCurveNode(conventionStr, timeStr, label, quoteKey, spread);
+    }
+    if ("FRA".equals(typeStr) || "Fra".equals(typeStr)) {
+      return curveFraCurveNode(conventionStr, timeStr, label, quoteKey, spread);
+    }
+    if ("OIS".equals(typeStr) || "FixedOvernightSwap".equals(typeStr)) {
+      return curveFixedOvernightCurveNode(conventionStr, timeStr, label, quoteKey, spread);
+    }
+    if ("IRS".equals(typeStr) || "FixedIborSwap".equals(typeStr)) {
+      return curveFixedIborCurveNode(conventionStr, timeStr, label, quoteKey, spread);
+    }
+    if ("BAS".equals(typeStr) || "IborIborSwap".equals(typeStr)) {
+      return curveIborIborCurveNode(conventionStr, timeStr, label, quoteKey, spread);
+    }
+    throw new IllegalArgumentException(Messages.format("Invalid curve node type: {}", typeStr));
+  }
+
+  private static CurveNode curveIborFixingDepositCurveNode(
+      String conventionStr,
+      String timeStr,
+      String label,
+      QuoteKey quoteKey,
+      double spread) {
+
+    IborFixingDepositConvention convention = IborFixingDepositConvention.of(conventionStr);
+    IborFixingDepositTemplate template = IborFixingDepositTemplate.of(
+        convention.getIndex().getTenor().getPeriod(), convention);
+    return IborFixingDepositCurveNode.of(template, quoteKey, spread);
+  }
+
+  private static CurveNode curveFraCurveNode(
+      String conventionStr,
+      String timeStr,
+      String label,
+      QuoteKey quoteKey,
+      double spread) {
+
+    Matcher matcher = FRA_TIME_REGEX.matcher(timeStr.toUpperCase(Locale.ENGLISH));
+    if (!matcher.matches()) {
+      throw new IllegalArgumentException(Messages.format("Invalid time format for FRA: {}", timeStr));
```

Comment:
Is there an easy way to express what a valid time format would look like? It would be good to add that to the error message.